src/eth/AccountsService.js

```diff
@@ -89,6 +89,25 @@ export default class AccountsService extends PublicService {
       });
     }
 
+    // Detect account change and automatically switch active account if
+    // autoSwitch flag set (Useful if using a browser wallet like MetaMask)
+    // See: https://github.com/MetaMask/faq/blob/master/DEVELOPERS.md#ear-listening-for-selected-account-changes
+    if (otherSettings.autoSwitch && type === AccountType.BROWSER) {
+      const accountCheckHandle = setInterval(async () => {
+        const activeBrowserAddress = window.web3.eth.defaultAccount.toLowerCase();
```

Comment:
If you pull the async function out as something that takes `accountService` and `web3` arguments, you can kill two birds with one stone—you can reduce the complexity, so you don’t need to change eslintrc, and you can write a test easily with a mock accountService and mock web3.

Reply:
There’s also one edge case: if the maker instance’s current account is not a browser-type one, e.g. a hardware wallet, I think you should add the new browser account but not switch to it, because in this case I think we’d want to keep using the hardware wallet.

Reply:
> There’s also one edge case: if the maker instance’s current account is not a browser-type one

Good point. I've updated it so that it will now only monitor for active account changes if the SDK's current account type is `browser`.